.cicd/build.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+set -eo pipefail
+. ./.cicd/helpers/general.sh
+
+fold-execute mkdir -p $BUILD_DIR
+
+if [[ $(uname) == 'Darwin' ]]; then
+
+    cd $BUILD_DIR
+    [[ $TRAVIS == true ]] && fold-execute ccache -s
+    fold-execute cmake -DCMAKE_BUILD_TYPE=Release ..
+    fold-execute make -j$JOBS
+
+else # Linux
+
+    MOUNTED_DIR='/workdir'
+    ARGS=${ARGS:-"--rm -v $(pwd):$MOUNTED_DIR"}
+
+    . $HELPERS_DIR/docker-hash.sh
+
+    PRE_COMMANDS=". $MOUNTED_DIR/.cicd/helpers/logging.sh"
+    COMMANDS="cd $MOUNTED_DIR/build && fold-execute cmake -DCMAKE_TOOLCHAIN_FILE=$MOUNTED_DIR/.cicd/helpers/clang.make -DENABLE_TESTS=ON -DCMAKE_BUILD_TYPE=Release .. && fold-execute make -j$JOBS"
```

Comment:
The `PRE_COMMANDS` and `COMMANDS` pattern obfuscates the code too much. It is not clear to me what advantages this provides, and I think we should hard-code all of these commands.